### DISCUSSION TITLE
intrinsic valuation time series

### DISCUSSION THREAD

**Discussion by nexdep** [JerBouma/FinanceToolkit#135] [Q&A] [answered]
Hello,
I am trying to calculate the DCF evaluation over time but the intrinsic valuation method seems to return it only for the latest statement.
Is there a way to get the time series? 
thanks for the help

**Comment by JerBouma** [comment, marked as answer]
Hi! The cash flow projection is based on the cashflow on the "Free Cash Flow", "Operating Cash Flow", "Change in Working Capital" or "Capital Expenditure" from the latest financial statements and then projects it forward given the growth rate, perpetual growth rate and WACC. So by setting periods to 1, you project one year forward given the growth rates and WACC.

You can obtain a timeseries by looping over these periods:

<img width="787" alt="image" src="https://github.com/JerBouma/FinanceToolkit/assets/46355364/900c1c32-2083-4fc9-9ee0-e93da3edb81e">

Do keep in mind that DCF models are highly sensitive to user input and can quickly become unreliable when market conditions change.

**Reply by nexdep** [reply]
Hi, thanks for taking the time to reply and for the library - I am just studying it now but it seems a great project.

I wasn't clear enough in my question, I'd like to calculate the how the DCF at five year evaluation changed depending on the previous financials. so not to change the projecting period parameter but just study its past evolution.

I tried to look at the source code but I am not sure it is natively implemented

tx

**Reply by nexdep** [reply]
bump :)

**Reply by JerBouma** [reply]
I'd recommend to look into the code itself. I've set it up in such a manner that you can use the actual model used to come to this conclusion as well: https://github.com/JerBouma/FinanceToolkit/blob/833ed9b2b1f19056c9e58ec1bf2793d728949cd4/financetoolkit/models/intrinsic_model.py#L9-L105

From there, and with all the data that is available in the Finance Toolkit you should be able to calculate what you are looking for. I can't create specialized functionality for each request so that is why I have seperated model from controller. E.g. you can directly access this function through `from financetoolkit.models import intrinsic_model`.

**Reply by nexdep** [reply]
Hi, thanks a lot for the answer - I'll try to go in that direction
